Drop Vector::append(T*, size_t) and use Vector::append(std::span<const T>) instead

cdumez opened the pull request:
#### c904754b70e17186e32cafafc5d82dd7bb925741
<pre>
Drop Vector::append(T*, size_t) and use Vector::append(std::span&lt;const T&gt;) instead
<a href="https://bugs.webkit.org/show_bug.cgi?id=271118">https://bugs.webkit.org/show_bug.cgi?id=271118</a>

Reviewed by Darin Adler and Brent Fulgham.

Drop Vector::append(T*, size_t) and use Vector::append(std::span&lt;const T&gt;) instead.
This is part of an effort to use std::span more in the codebase, for better security.

* Source/JavaScriptCore/API/ObjcRuntimeExtras.h:
(forEachProtocolImplementingProtocol):
* Source/JavaScriptCore/assembler/AbstractMacroAssembler.h:
(JSC::AbstractMacroAssembler::JumpList::append):
* Source/JavaScriptCore/bytecode/ObjectPropertyConditionSet.cpp:
(JSC::ObjectPropertyConditionSet::mergedWith const):
* Source/JavaScriptCore/jsc.cpp:
(runWithOptions):
* Source/JavaScriptCore/parser/Lexer.cpp:
(JSC::Lexer&lt;CharacterType&gt;::parseIdentifierSlowCase):
(JSC::Lexer&lt;T&gt;::parseTemplateLiteral):
* Source/JavaScriptCore/parser/Lexer.h:
(JSC::Lexer::append16):
* Source/JavaScriptCore/runtime/IntlLocale.cpp:
(JSC::LocaleIDBuilder::overrideLanguageScriptRegion):
* Source/JavaScriptCore/runtime/IntlObject.cpp:
(JSC::canonicalizeUnicodeExtensionsAfterICULocaleCanonicalization):
(JSC::addScriptlessLocaleIfNeeded):
* Source/JavaScriptCore/runtime/StringPrototype.cpp:
(JSC::illFormedIndex):
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/tools/JSDollarVM.cpp:
* Source/JavaScriptCore/wasm/WasmBBQPlan.h:
* Source/JavaScriptCore/wasm/WasmEntryPlan.cpp:
(JSC::Wasm::EntryPlan::parseAndValidateModule):
* Source/JavaScriptCore/wasm/WasmEntryPlan.h:
* Source/JavaScriptCore/wasm/WasmIPIntPlan.cpp:
(JSC::Wasm::IPIntPlan::IPIntPlan):
* Source/JavaScriptCore/wasm/WasmLLIntPlan.cpp:
(JSC::Wasm::LLIntPlan::LLIntPlan):
* Source/JavaScriptCore/wasm/WasmStreamingCompiler.h:
* Source/JavaScriptCore/wasm/WasmStreamingParser.cpp:
(JSC::Wasm::StreamingParser::consume):
(JSC::Wasm::StreamingParser::consumeVarUInt32):
(JSC::Wasm::StreamingParser::addBytes):
(JSC::Wasm::StreamingParser::finalize):
* Source/JavaScriptCore/wasm/WasmStreamingParser.h:
(JSC::Wasm::StreamingParser::addBytes):
* Source/WTF/wtf/StreamBuffer.h:
(WTF::StreamBuffer::append):
* Source/WTF/wtf/TrailingArray.h:
(WTF::TrailingArray::span const):
* Source/WTF/wtf/URL.cpp:
(WTF::appendEncodedHostname):
* Source/WTF/wtf/URLParser.cpp:
(WTF::URLParser::appendToASCIIBuffer):
(WTF::URLParser::utf8PercentEncode):
(WTF::URLParser::utf8QueryEncode):
(WTF::URLParser::copyASCIIStringUntil):
(WTF::URLParser::syntaxViolation):
(WTF::URLParser::parse):
(WTF::URLParser::appendNumberToASCIIBuffer):
(WTF::URLParser::serializeIPv6):
(WTF::URLParser::domainToASCII):
(WTF::URLParser::parseHostAndPort):
* Source/WTF/wtf/URLParser.h:
(WTF::URLParser::appendToASCIIBuffer): Deleted.
* Source/WTF/wtf/Vector.h:
(WTF::Vector::subspan const):
(WTF::Vector::tryAppend):
(WTF::Vector::append):
(WTF::Vector::appendList):
(WTF::Malloc&gt;::append):
(WTF::Malloc&gt;::appendVector):
* Source/WTF/wtf/cf/VectorCF.h:
* Source/WTF/wtf/cocoa/FileSystemCocoa.mm:
(WTF::FileSystemImpl::openTemporaryFile):
* Source/WTF/wtf/text/AtomString.h:
* Source/WTF/wtf/text/SuperFastHash.h:
(WTF::SuperFastHash::addCharacters):
(WTF::SuperFastHash::computeHash):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::String::charactersWithoutNullTermination const):
* Source/WebCore/Modules/indexeddb/server/IDBSerialization.cpp:
(WebCore::writeLittleEndian):
(WebCore::encodeKey):
(WebCore::decodeKey):
* Source/WebCore/Modules/mediastream/SFrameUtils.cpp:
(WebCore::toRbsp):
(WebCore::computeVP8PrefixBuffer):
* Source/WebCore/Modules/mediastream/STUNMessageParsing.cpp:
(WebCore::WebRTC::getSTUNOrTURNMessageLengths):
(WebCore::WebRTC::extractSTUNOrTURNMessages):
* Source/WebCore/Modules/mediastream/STUNMessageParsing.h:
* Source/WebCore/Modules/webauthn/cbor/CBORReader.cpp:
(cbor::CBORReader::readBytes):
* Source/WebCore/Modules/webauthn/cbor/CBORWriter.cpp:
(cbor::CBORWriter::encodeCBOR):
* Source/WebCore/Modules/webauthn/fido/FidoHidPacket.cpp:
(fido::FidoHidInitPacket::getSerializedData const):
(fido::FidoHidContinuationPacket::getSerializedData const):
* Source/WebCore/Modules/webauthn/fido/FidoParsingUtils.cpp:
(fido::getInitPacketData):
(fido::getContinuationPacketData):
* Source/WebCore/Modules/webauthn/fido/U2fCommandConstructor.cpp:
(fido::WebCore::constructU2fSignCommand):
* Source/WebCore/Modules/websockets/WebSocketFrame.cpp:
(WebCore::WebSocketFrame::makeFrameData):
* Source/WebCore/Modules/websockets/WebSocketHandshake.cpp:
(WebCore::WebSocketHandshake::readHTTPHeaders):
* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(PAL::appendDecimal):
* Source/WebCore/PAL/pal/text/TextCodecICU.cpp:
(PAL::TextCodecICU::encode const):
* Source/WebCore/PAL/pal/text/TextCodecLatin1.cpp:
(PAL::encodeComplexWindowsLatin1):
* Source/WebCore/PAL/pal/text/TextCodecUserDefined.cpp:
(PAL::encodeComplexUserDefined):
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::supportsPressAction const):
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::handleResponseOnStreamingAction):
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::writeLittleEndian):
(WebCore::writeLittleEndian&lt;uint8_t&gt;):
(WebCore::CloneSerializer::serialize):
(WebCore::CloneSerializer::dumpImageBitmap):
(WebCore::CloneSerializer::dumpIfTerminal):
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::read):
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::SubtleCrypto::wrapKey):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmECDSAMac.cpp:
(WebCore::signECDSA):
(WebCore::verifyECDSA):
* Source/WebCore/crypto/cocoa/CryptoKeyECMac.cpp:
(WebCore::CryptoKeyEC::platformExportSpki const):
(WebCore::CryptoKeyEC::platformImportPkcs8):
(WebCore::CryptoKeyEC::platformExportPkcs8 const):
* Source/WebCore/crypto/cocoa/CryptoKeyOKPCocoa.cpp:
(WebCore::CryptoKeyOKP::exportSpki const):
(WebCore::CryptoKeyOKP::exportPkcs8 const):
* Source/WebCore/crypto/cocoa/CryptoKeyRSAMac.cpp:
(WebCore::CryptoKeyRSA::exportSpki const):
(WebCore::CryptoKeyRSA::exportPkcs8 const):
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCbcCfbParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCtrParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesGcmParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHkdfParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmPbkdf2Params.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaKeyGenParams.h:
(WebCore::CryptoAlgorithmRsaKeyGenParams::publicExponentVector const):
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaOaepParams.h:
* Source/WebCore/dom/DatasetDOMStringMap.cpp:
(WebCore::convertPropertyNameToAttributeName):
* Source/WebCore/fileapi/BlobBuilder.cpp:
(WebCore::BlobBuilder::append):
* Source/WebCore/history/HistoryItem.cpp:
(WebCore::HistoryItem::showTreeWithIndent const):
* Source/WebCore/html/parser/HTMLToken.h:
(WebCore::HTMLToken::appendToComment):
* Source/WebCore/loader/TextResourceDecoder.cpp:
(WebCore::TextResourceDecoder::decode):
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::parseEventStreamLine):
* Source/WebCore/page/PageConsoleClient.cpp:
(WebCore::PageConsoleClient::addMessage):
(WebCore::PageConsoleClient::messageWithTypeAndLevel):
* Source/WebCore/platform/SharedBuffer.cpp:
(WebCore::combineSegmentsData):
(WebCore::FragmentedSharedBuffer::getContiguousData const):
(WebCore::FragmentedSharedBuffer::read const):
* Source/WebCore/platform/SharedStringHash.cpp:
(WebCore::findSlashDotDotSlash):
(WebCore::findSlashSlash):
(WebCore::findSlashDotSlash):
(WebCore::containsColonSlashSlash):
(WebCore::cleanSlashDotDotSlashes):
(WebCore::mergeDoubleSlashes):
(WebCore::cleanSlashDotSlashes):
(WebCore::cleanPath):
(WebCore::needsTrailingSlash):
(WebCore::computeSharedStringHashInline):
(WebCore::computeSharedStringHash):
(WebCore::computeVisitedLinkHash):
* Source/WebCore/platform/SharedStringHash.h:
* Source/WebCore/platform/audio/MultiChannelResampler.cpp:
(WebCore::MultiChannelResampler::provideInputForChannel):
* Source/WebCore/platform/audio/SincResampler.cpp:
(WebCore::SincResampler::processBuffer):
(WebCore::SincResampler::process):
* Source/WebCore/platform/graphics/cg/ImageBufferUtilitiesCG.cpp:
(WebCore::encode):
(WebCore::encodeToVector):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::WebMParser::AudioTrackData::consumeFrameData):
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.h:
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.mm:
(WebCore::cookieFromVorbisCodecPrivate):
(WebCore::createVorbisAudioInfo):
* Source/WebCore/platform/network/FormData.cpp:
(WebCore::FormData::flatten const):
* Source/WebCore/platform/network/FormDataBuilder.cpp:
(WebCore::FormDataBuilder::append):
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::parseHTTPHeader):
(WebCore::parseHTTPRequestBody):
* Source/WebCore/platform/network/HTTPParsers.h:
* Source/WebCore/platform/network/SynchronousLoaderClient.cpp:
(WebCore::SynchronousLoaderClient::didReceiveData):
* Source/WebCore/platform/network/curl/CurlMultipartHandle.cpp:
(WebCore::CurlMultipartHandle::parseHeadersIfPossible):
* Source/WebCore/platform/text/QuotedPrintable.cpp:
(WebCore::quotedPrintableEncode):
* Source/WebCore/rendering/svg/SVGTextChunk.cpp:
(WebCore::SVGTextChunk::SVGTextChunk):
* Source/WebCore/svg/SVGToOTFFontConversion.cpp:
(WebCore::SVGToOTFFontConverter::appendOS2Table):
* Source/WebCore/testing/MockContentFilter.cpp:
(WebCore::MockContentFilter::maybeDetermineStatus):
* Source/WebCore/xml/parser/XMLDocumentParser.h:
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::PendingCallbacks::appendCharactersCallback):
(WebCore::XMLDocumentParser::characters):
(WebCore::charactersHandler):
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::sendReplyToSynchronousRequest):
* Source/WebKit/NetworkProcess/webrtc/LibWebRTCSocketClient.cpp:
(WebKit::LibWebRTCSocketClient::sendTo):
* Source/WebKit/NetworkProcess/webrtc/LibWebRTCSocketClient.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::sendToSocket):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.mm:
(WebKit::NetworkRTCTCPSocketCocoa::createMessageBuffer):
(WebKit::NetworkRTCTCPSocketCocoa::sendTo):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
(WebKit::NetworkRTCUDPSocketCocoa::sendTo):
(WebKit::NetworkRTCUDPSocketCocoaConnections::sendTo):
* Source/WebKit/Shared/API/APIData.h:
(API::Data::data const):
(API::Data::bytes const):
(API::Data::dataReference const): Deleted.
* Source/WebKit/Shared/API/APIData.serialization.in:
* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
(-[WKRemoteObjectDecoder decodeBytesForKey:returnedLength:]):
* Source/WebKit/Shared/API/c/WKData.cpp:
(WKDataGetBytes):
* Source/WebKit/Shared/APIWebArchive.mm:
(API::WebArchive::WebArchive):
* Source/WebKit/Shared/APIWebArchiveResource.mm:
(API::WebArchiveResource::WebArchiveResource):
* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:
(WebKit::SandboxExtension::createHandleForTemporaryFile):
* Source/WebKit/Shared/Cocoa/WKNSData.mm:
(-[WKNSData bytes]):
* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::fileContents):
(WebKit::compileAndCacheSandboxProfile):
* Source/WebKit/UIProcess/API/C/WKNotificationManager.cpp:
(WKNotificationManagerProviderDidClickNotification_b):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageLoadData):
(WKPageLoadDataWithUserData):
(restoreFromSessionState):
* Source/WebKit/UIProcess/API/C/WKSessionStateRef.cpp:
(WKSessionStateCreateFromData):
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.cpp:
(WebKit::WebNotificationManagerProxy::providerDidCloseNotifications):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didChooseFilesForOpenPanelWithDisplayStringAndIcon):
(WebKit::WebPageProxy::getWebCryptoMasterKey):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resumeDownload):
* Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp:
(WebKit::HistoryEntryDataDecoder::operator&gt;&gt;):
* Source/WebKit/UIProcess/mac/WKPrintingView.mm:
(-[WKPrintingView _preparePDFDataForPrintingOnSecondaryThread]):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageEditorClient.cpp:
(WebKit::InjectedBundlePageEditorClient::getPasteboardDataForRange):
* Source/WebKit/WebProcess/InjectedBundle/mac/InjectedBundleMac.mm:
(WebKit::createUnarchiver):
* Source/WebKit/WebProcess/Plugins/PDF/PDFIncrementalLoader.mm:
(WebKit::ByteRangeRequest::addData):
(WebKit::PDFPluginStreamLoaderClient::didReceiveData):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::completeTextManipulation):
* Source/WebKitLegacy/WebCoreSupport/SocketStreamHandleClient.h:
* Source/WebKitLegacy/WebCoreSupport/SocketStreamHandleImpl.cpp:
(WebCore::removeTerminationCharacters):
(WebCore::cookieDataForHandshake):
(WebCore::SocketStreamHandleImpl::platformSendHandshake):
* Source/WebKitLegacy/WebCoreSupport/SocketStreamHandleImplCFNet.cpp:
(WebCore::SocketStreamHandleImpl::readStreamCallback):
* Source/WebKitLegacy/WebCoreSupport/WebCryptoClient.mm:
(WebCryptoClient::wrapCryptoKey const):
(WebCryptoClient::unwrapCryptoKey const):
* Source/WebKitLegacy/WebCoreSupport/WebSocketChannel.cpp:
(WebCore::WebSocketChannel::didReceiveSocketStreamData):
(WebCore::WebSocketChannel::appendToBuffer):
* Source/WebKitLegacy/WebCoreSupport/WebSocketChannel.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebVisitedLinkStore.mm:
(WebVisitedLinkStore::addVisitedLink):
* Tools/DumpRenderTree/PixelDumpSupport.cpp:
(convertChecksumToPNGComment):
* Tools/TestWebKitAPI/Tests/WebCore/ASN1Utilities.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/CtapPinTest.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/CtapRequestTest.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/CtapResponseTest.cpp:
(TestWebKitAPI::getTestAttestedCredentialDataBytes):
(TestWebKitAPI::getTestAttestationObjectBytes):
(TestWebKitAPI::getTestCorruptedSignResponse):
(TestWebKitAPI::getTestU2fRegisterResponse):
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/MediaLoading.mm:
(TestWebKitAPI::testVideoBytes):
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsiteDataStoreCustomPaths.mm:
(respondToRangeRequests):
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
(TestWebKitAPI::vectorFromData):
(TestWebKitAPI::appendUTF8ToVector):
(TestWebKitAPI::H2::Connection::receive const):
* Tools/WebKitTestRunner/PixelDumpSupport.cpp:
(convertChecksumToPNGComment):

Canonical link: <a href="https://commits.webkit.org/276319@main">https://commits.webkit.org/276319@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0a419535ef82363eddab5266cdfa4640953d2801

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44308 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23375 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46743 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46957 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40337 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46612 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27351 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20772 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36495 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44884 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20419 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38156 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/17534 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/17902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39269 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2356 "Built successfully") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40490 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39552 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48565 "Built successfully") | 
| [⏳ 🛠 🧪 jsc-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-arm64-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19286 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/15852 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/43399 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/44357 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/20648 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/38427 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/42129 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/20976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/50946 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6091 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20273 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10315 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/JavaScriptCore/inspector/remote/socket/RemoteInspectorMessageParser.cpp:
```diff
@@ -59,12 +59,12 @@ Vector<uint8_t> MessageParser::createMessage(const uint8_t* data, size_t size)
+void MessageParser::pushReceivedData(std::span<const uint8_t> data)
 {
-    if (!data || !size || !m_listener)
+    if (!data.data() || !data.size() || !m_listener)
```
Doesn’t seem like we need to check `data.data()` for null. I understand the old code did that.

darinadler commented on Source/JavaScriptCore/runtime/IntlObject.cpp:
```diff
@@ -384,15 +383,14 @@ Vector<char, 32> canonicalizeUnicodeExtensionsAfterICULocaleCanonicalization(Vec
     unsigned remainingStart = extensionIndex + extensionLength;
-    unsigned remainingLength = buffer.size() - remainingStart;
-    result.append(buffer.data() + remainingStart, remainingLength);
+    result.append(buffer.subspan(remainingStart));
```
We don’t need the local variable any more:
```
result.append(buffer.subspan(extensionIndex + extensionLength));
```

darinadler commented on Source/WTF/wtf/TrailingArray.h:
```diff
@@ -96,6 +96,7 @@ class TrailingArray {
 
     pointer data() { return bitwise_cast<T*>(bitwise_cast<uint8_t*>(static_cast<Derived*>(this)) + offsetOfData()); }
     const_pointer data() const { return bitwise_cast<const T*>(bitwise_cast<const uint8_t*>(static_cast<const Derived*>(this)) + offsetOfData()); }
+    std::span<const T> span() const { return { data(), size() }; }
```
Do we want a non-const overload as well?

cdumez replied:
Will add, to be consistent with `data()`

darinadler commented on Source/WTF/wtf/URLParser.cpp:
```diff
@@ -489,8 +489,7 @@ void URLParser::percentEncodeByte(uint8_t byte)
 
-const char replacementCharacterUTF8PercentEncoded[10] = "%EF%BF%BD";
-const size_t replacementCharacterUTF8PercentEncodedLength = sizeof(replacementCharacterUTF8PercentEncoded) - 1;
+static constexpr ASCIILiteral replacementCharacterUTF8PercentEncoded = "%EF%BF%BD"_s;
```
We could just write `constexpr auto`, don’t need `static`.

darinadler commented on Source/WTF/wtf/URLParser.cpp:
```diff
@@ -1378,7 +1377,7 @@ void URLParser::parse(const CharacterType* input, const unsigned length, const U
                 ++c;
                 copyURLPartsUntil(base, URLPart::SchemeEnd, c, nonUTF8QueryEncoding);
-                appendToASCIIBuffer("://", 3);
+                appendToASCIIBuffer("://"_s.span8());
```
Maybe we should add a `_span` that we can use instead of `_s.span8()`?

cdumez replied:
Ok, I can add this:
```
constexpr std::span<const LChar> operator"" _span(const char* characters, size_t n)
{
#if ASSERT_ENABLED
    for (size_t i = 0; i < n; ++i)
        ASSERT_UNDER_CONSTEXPR_CONTEXT(isASCII(characters[i]));
#endif
    return std::span { bitwise_cast<const LChar*>(characters), n };
}
```

How does it look?

darinadler replied:
I think it’s great. You can imagine wanting one that does `std::span<const char>`, but for our uses we clearly want `LChar` or `uint8_t`, which happen to both be the same at the moment.

darinadler commented on Source/WTF/wtf/URLParser.cpp:
```diff
@@ -2851,7 +2850,7 @@ auto URLParser::parseHostAndPort(CodePointIterator<CharacterType> iterator) -> H
         if (isError)
             return HostParsingResult::InvalidHost;
-        utf8Encoded.append(buffer, offset);
+        utf8Encoded.append(std::span { buffer, static_cast<size_t>(offset) });
```
We could change `offset` to `size_t` instead.

darinadler commented on Source/WTF/wtf/URLParser.cpp:
```diff
@@ -2576,7 +2575,7 @@ template<typename CharacterType> std::optional<URLParser::LCharBuffer> URLParser
         UNUSED_PARAM(numCharactersConverted);
 #endif // ASSERT_ENABLED
-        ascii.append(hostnameBuffer, numCharactersConverted);
+        ascii.append(std::span { hostnameBuffer, static_cast<size_t>(numCharactersConverted) });
```
We could change `numCharactersConverted` to `size_t` instead.

darinadler commented on Source/WTF/wtf/Vector.h:
```diff
@@ -806,6 +806,11 @@ class Vector : private VectorBuffer<T, inlineCapacity, Malloc> {
         return { span().subspan(offset, length) };
     }
 
+    std::span<const T> subspan(size_t offset, size_t length = std::dynamic_extent) const
```
I don’t think `std::dynamic_extent` is the right thing here. We are using this because it’s the maximum value, not because it’s the dynamic extent indicator.

cdumez replied:
Hmmm I'm merely forwarding the call to std::span::subspan() which DOES use `std::dynamic_extent`:
https://en.cppreference.com/w/cpp/container/span/subspan

At least I'm consistent with STL :)

darinadler replied:
Yes, I guess I have a bone to pick with the creators of the standard library, not with you.

brentfulgham commented on Source/JavaScriptCore/jsc.cpp:
```diff
@@ -3699,7 +3699,7 @@ static void runWithOptions(GlobalObject* globalObject, CommandLine& options, boo
             fileName = String::fromLatin1(scripts[i].argument);
             if (scripts[i].strictMode == Script::StrictMode::Strict)
-                scriptBuffer.append("\"use strict\";\n", strlen("\"use strict\";\n"));
+                scriptBuffer.append("\"use strict\";\n"_s.span8());
```
Clever!

brentfulgham commented on Source/WTF/wtf/URLParser.h:
```diff
@@ -124,8 +124,7 @@ class URLParser {
     void appendToASCIIBuffer(char32_t);
-    void appendToASCIIBuffer(const char*, size_t);
-    void appendToASCIIBuffer(const LChar* characters, size_t size) { appendToASCIIBuffer(reinterpret_cast<const char*>(characters), size); }
+    void appendToASCIIBuffer(std::span<const LChar>);
```
Do we not need the dedicated `char*` overload? If not, why do we ever declare `LChar` arguments and `char` arguments?

cdumez replied:
We don't need it because it is not getting called anymore. `LChar` and `char` are still different. `LChar` is currently a `uint8_t`.

brentfulgham commented on Source/WTF/wtf/text/SuperFastHash.h:
```diff
@@ -156,6 +163,12 @@ class SuperFastHash {
+    template<typename T, typename Converter = DefaultConverter>
+    static constexpr unsigned computeHash(std::span<const T> data)
+    {
+        return StringHasher::finalize(computeHashImpl<T, Converter>(data.data(), data.size()));
```
Obviously it's time for `computeHashImpl` to learn about `span`! :-)

cdumez replied:
I agree. I'll follow-up though as this patch is already large. There is also a very slight performance concern since this is likely hot and std::span does bounds checking.

darinadler commented on Source/WebCore/Modules/mediastream/SFrameUtils.cpp:
```diff
@@ -174,7 +174,7 @@ void toRbsp(Vector<uint8_t>& frame, size_t offset)
     Vector<uint8_t> newFrame;
     newFrame.reserveInitialCapacity(frame.size() + count);
-    newFrame.append(frame.data(), offset);
+    newFrame.append(frame.subspan(0, offset));
```
`frame.first`

cdumez replied:
Same comment as earlier. `frame` is a Vector, not a span.

darinadler commented on Source/WebCore/accessibility/AccessibilityObject.cpp:
```diff
@@ -3009,7 +3009,7 @@ bool AccessibilityObject::supportsPressAction() const
         const auto& children = candidate->children();
         if (size_t maxChildrenToTake = std::min(static_cast<size_t>(halfSearchLimit), candidatesLeftUntilLimit))
-            candidates.append(children.span().subspan(0, std::min(children.size(), maxChildrenToTake)));
+            candidates.append(children.subspan(0, std::min(children.size(), maxChildrenToTake)));
```
`children.first`

cdumez replied:
children is a Vector, not a span.

darinadler commented on Source/WebCore/crypto/cocoa/CryptoKeyECMac.cpp:
```diff
@@ -391,25 +391,25 @@ Vector<uint8_t> CryptoKeyEC::platformExportPkcs8() const
     addEncodedASN1Length(result, publicKeySize);
     result.append(InitialOctet);
-    result.append(keyBytes.data(), publicKeySize - 1);
+    result.append(keyBytes.subspan(0, publicKeySize - 1));
```
`keyBytes.first`

cdumez replied:
keyBytes is a Vector, not a span.

darinadler commented on Source/JavaScriptCore/runtime/IntlObject.cpp:
```diff
@@ -354,16 +354,15 @@ Vector<char, 32> canonicalizeUnicodeExtensionsAfterICULocaleCanonicalization(Vec
 
-    Vector<char, 32> result;
-    result.append(buffer.data(), extensionIndex + 2); // "-u" is included.
+    Vector<char, 32> result(buffer.subspan(0, extensionIndex + 2)); // "-u" is included.
```
`buffer.first`

cdumez replied:
The issue is that `buffer` is a Vector, not a span. I did add `subspan()` to Vector in this PR. However, I didn't add `first()` or `last()`.

darinadler replied:
Makes sense, in fact I think I even figured that out later; please ignore my incorrect suggestions.

brentfulgham commented on Source/WebCore/PAL/pal/text/TextCodecUserDefined.cpp:
```diff
@@ -68,7 +68,7 @@ static Vector<uint8_t> encodeComplexUserDefined(StringView string, UnencodableHa
             // No way to encode this character with x-user-defined.
             UnencodableReplacementArray replacement;
```
Maybe `UnencodableReplacementArray` needs a `span<>` operator/accessor.

cdumez replied:
```
using UnencodableReplacementArray = std::array<char, 32>;
```

So I can actually call ` std::span { replacement }` directly. I'll fix.

cdumez replied:
Scratch that. Even though `UnencodableReplacementArray` is a std::array, I cannot simply call `std::span { replacement }` because `replacement.size()` may differ from `replacementLength`. I don't think I can simplify here.

brentfulgham replied:
Ah, that makes sense. Thank you for explaining.

darinadler commented on Source/WebCore/Modules/webauthn/fido/FidoParsingUtils.cpp:
```diff
@@ -34,19 +34,15 @@ namespace fido {
-    Vector<uint8_t> result;
-    result.append(data.data(), data.size() > kHidInitPacketDataSize ? kHidInitPacketDataSize : data.size());
-    return result;
+    return data.subvector(0, data.size() > kHidInitPacketDataSize ? kHidInitPacketDataSize : data.size());
```
`std::min(kHidInitPacketDataSize, data.size())`

brentfulgham commented on Source/WebCore/bindings/js/SerializedScriptValue.cpp:
```diff
@@ -965,7 +966,7 @@ static std::optional<Vector<uint8_t>> unwrapCryptoKey(JSGlobalObject* lexicalGlo
 template <typename T> static void writeLittleEndian(Vector<uint8_t>& buffer, T value)
 {
-    buffer.append(reinterpret_cast<uint8_t*>(&value), sizeof(value));
+    buffer.append(std::span { reinterpret_cast<uint8_t*>(&value), sizeof(value) });
```
You've used this pattern a few times in this patch. I wonder if a `toSpan<>` template would clean thing up a bit more?

darinadler commented on Source/WebCore/Modules/webauthn/fido/FidoParsingUtils.cpp:
```diff
@@ -34,19 +34,15 @@ namespace fido {
-    Vector<uint8_t> result;
-    result.append(data.data() + beginPosition, data.size() > kHidContinuationPacketDataSize + beginPosition ? kHidContinuationPacketDataSize : data.size() - beginPosition);
-    return result;
+    return data.subspan(beginPosition, data.size() > kHidContinuationPacketDataSize + beginPosition ? kHidContinuationPacketDataSize : data.size() - beginPosition);
```
`std::min(kHidContinuationPacketDataSize, data.size() - beginPosition)`

brentfulgham commented on Source/WebCore/bindings/js/SerializedScriptValue.cpp:
```diff
@@ -2377,9 +2378,9 @@ class CloneSerializer : public CloneBase {
-        write(CFDataGetBytePtr(data.get()), dataLength);
+        auto span = toSpan(data.get());
+        write(static_cast<uint32_t>(span.size()));
+        write(span);
```
I'm not sure this trip through `span` makes this code better. Though, if `write()` had an overload that took a `span`, that might make it better.

brentfulgham replied:
Oh, I guess it does, but only for a `span<const uint8_t>`. Maybe adding another of `span<const uint32_t>` would be too much.

cdumez replied:
I don't understand the feedback here. write() does take a span and I am passing it a span. I merely constructed the span a little earlier to avoid calling `CFDataGetLength()` twice, once explicitly and a second time inside `toSpan()`.

brentfulgham replied:
I was just pointing out that since a `span` knows its size, the serializer needing to accept a size, then pass the span (which knows the size) seems like extra effort. Especially if every SerializedScriptValue user with a span has to know to send the two values independently.

darinadler commented on Source/WebCore/Modules/websockets/WebSocketFrame.cpp:
```diff
@@ -162,7 +162,7 @@ void WebSocketFrame::makeFrameData(Vector<uint8_t>& frameData)
         }
         ASSERT(!remaining);
-        frameData.append(extendedPayloadLength, 8);
+        frameData.append(std::span { extendedPayloadLength });
```
Does this work correctly without changing `extendedPayloadLength` to `std::array`?

cdumez replied:
As far as I know, it works with C++20. I have also tried the following:
```
    uint8_t extendedPayloadLength[8];
    std::span sp { extendedPayloadLength };
    std::cout << sp.size() << std::endl;
```
and it prints out 8. so I believe it works.

brentfulgham commented on Source/WebCore/crypto/gcrypt/CryptoKeyECGCrypt.cpp:
```diff
@@ -163,7 +163,7 @@ RefPtr<CryptoKeyEC> CryptoKeyEC::platformImportJWKPublic(CryptoAlgorithmIdentifi
     Vector<uint8_t> q;
     q.reserveInitialCapacity(curveUncompressedPointSize(curve));
-    q.append(CryptoConstants::s_ecUncompressedFormatLeadingByte.data(), CryptoConstants::s_ecUncompressedFormatLeadingByte.size());
+    q.append(std::span { CryptoConstants::s_ecUncompressedFormatLeadingByte });
```
This is so much more readable!

darinadler commented on Source/WebCore/bindings/js/SerializedScriptValue.cpp:
```diff
@@ -3760,7 +3761,7 @@ class CloneDeserializer : public CloneBase {
             return false;
         if (m_ptr + size > m_end)
```
This existing code looks wrong from the point of view of overflow. It should be `if (size > m_end - m_ptr)`, which could be particularly important in 32-bit. Maybe we’d need type casts, though?

brentfulgham commented on Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
```diff
@@ -386,7 +386,7 @@ static inline void processUDPData(RetainPtr<nw_connection_t>&& nwConnection, Ref
     }
 
-    auto value = adoptNS(dispatch_data_create(data, size, nullptr, DISPATCH_DATA_DESTRUCTOR_DEFAULT));
+    RetainPtr value = adoptNS(dispatch_data_create(data.data(), data.size(), nullptr, DISPATCH_DATA_DESTRUCTOR_DEFAULT));
```
Why isn't this still `auto` -- doesn't `adoptNS` mean we get a `RetainPtr`?

cdumez replied:
We don't hide smart pointers behind `auto` anymore, to make it clear we're dealing with smart pointers. No behavior change though.

darinadler commented on Source/WebCore/platform/SharedStringHash.cpp:
```diff
@@ -192,39 +192,38 @@ static inline bool needsTrailingSlash(const CharacterType* characters, unsigned
-    if (!length || url.is8Bit())
-        return computeSharedStringHashInline(url.characters8(), length);
-    return computeSharedStringHashInline(url.characters16(), length);
+    if (!url.length() || url.is8Bit())
```
Can use `isEmpty()` instead of `length()`.

darinadler commented on Source/WebCore/platform/SharedStringHash.cpp:
```diff
@@ -192,39 +192,38 @@ static inline bool needsTrailingSlash(const CharacterType* characters, unsigned
+static ALWAYS_INLINE SharedStringHash computeSharedStringHashInline(const URL& base, std::span<const CharacterType> characters)
 {
-    if (!length)
+    if (!characters.size())
```
Could use `empty()` instead of `size()`.

darinadler commented on Source/WebCore/platform/SharedStringHash.cpp:
```diff
@@ -238,20 +237,20 @@ static ALWAYS_INLINE SharedStringHash computeSharedStringHashInline(const URL& b
 
     Vector<CharacterType, 512> buffer;
-    if (!length)
+    if (!characters.size())
```
Could use `empty()` instead of `size()`.

brentfulgham reviewed (approved):
Nice clean-up! r=me

darinadler commented on Source/WebCore/platform/graphics/opentype/OpenTypeUtilities.cpp:
```diff
@@ -189,7 +189,7 @@ void EOTHeader::appendBigEndianString(const BigEndianUShort* string, unsigned sh
 {
     unsigned short padding = 0;
-    m_buffer.append(reinterpret_cast<uint8_t*>(&padding), sizeof(padding));
+    m_buffer.append(std::span { reinterpret_cast<uint8_t*>(&padding), sizeof(padding) });
```
Could write this to avoid reinterpret_cast:
```
std::array<uint8_t, sizeof(unsigned short)> padding = { };
m_buffer.append(std::span { padding });
```

darinadler commented on Source/WebCore/platform/text/QuotedPrintable.cpp:
```diff
@@ -33,12 +33,13 @@
 static const size_t maximumLineLength = 76;
 
-static const char crlfLineEnding[] = "\r\n";
+static constexpr ASCIILiteral crlfLineEnding = "\r\n"_s;
```
As with the other places, I think it can be `constexpr auto`.

darinadler commented on Source/WebKit/Shared/API/APIData.h:
```diff
@@ -86,9 +86,9 @@ class Data : public ObjectImpl<API::Object::Type::Data> {
     }
 
-    const unsigned char* bytes() const { return m_span.data(); }
+    const unsigned char* data() const { return m_span.data(); }
```
Do we even need this function?

cdumez replied:
I seem to remember there are still a few call sites. I can look into removing it though.

darinadler commented on Source/WebKitLegacy/WebCoreSupport/WebCryptoClient.mm:
```diff
@@ -37,7 +38,7 @@
         NSData *keyData = CallUIDelegate(m_webView, selector);
         Vector<uint8_t> masterKey;
-        masterKey.append(static_cast<uint8_t*>(const_cast<void*>([keyData bytes])), [keyData length]);
+        masterKey.append(toSpan(keyData));
```
We should initialize to the span rather than appending.

darinadler commented on Source/WebKitLegacy/WebCoreSupport/WebCryptoClient.mm:
```diff
@@ -58,7 +59,7 @@
         Vector<uint8_t> masterKey;
         NSData *keyData = CallUIDelegate(m_webView, selector);
-        masterKey.append(static_cast<uint8_t*>(const_cast<void*>([keyData bytes])), [keyData length]);
+        masterKey.append(toSpan(keyData));
```
We should initialize to the span rather than appending.

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/MediaLoading.mm:
```diff
@@ -207,7 +206,7 @@ HTTPServer server([&](Connection connection) {
         Vector<uint8_t> vector;
-        vector.append(static_cast<const uint8_t*>(data.bytes), data.length);
+        vector.append(toSpan(data));
         return vector;
```
Can just `return toVector(data)`. Also I think this could just be a one-liner.

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/MediaLoading.mm:
```diff
@@ -134,9 +135,7 @@ HTTPServer manifestServer([&](Connection connection) mutable {
-    Vector<uint8_t> vector;
-    vector.append(static_cast<const uint8_t*>(data.bytes), data.length);
-    return vector;
+    return toSpan(data);
```
Should be `toVector` instead of `toSpan`. Also I think this could just be a one-liner.

darinadler commented on Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
```diff
@@ -348,16 +348,15 @@ static ASCIILiteral statusText(unsigned statusCode)
     __block Vector<uint8_t> request;
     dispatch_data_apply(content, ^bool(dispatch_data_t, size_t, const void* buffer, size_t size) {
-        request.append(static_cast<const char*>(buffer), size);
+        request.append(std::span { static_cast<const char*>(buffer), size });
```
Should be `const uint8_t*` instead of `const char*`.

cdumez committed:
Drop Vector::append(T*, size_t) and use Vector::append(std::span<const T>) instead
https://bugs.webkit.org/show_bug.cgi?id=271118

Reviewed by Darin Adler and Brent Fulgham.

Drop Vector::append(T*, size_t) and use Vector::append(std::span<const T>) instead.
This is part of an effort to use std::span more in the codebase, for better security.

* Source/JavaScriptCore/API/ObjcRuntimeExtras.h:
(forEachProtocolImplementingProtocol):
* Source/JavaScriptCore/assembler/AbstractMacroAssembler.h:
(JSC::AbstractMacroAssembler::JumpList::append):
* Source/JavaScriptCore/bytecode/ObjectPropertyConditionSet.cpp:
(JSC::ObjectPropertyConditionSet::mergedWith const):
* Source/JavaScriptCore/jsc.cpp:
(runWithOptions):
* Source/JavaScriptCore/parser/Lexer.cpp:
(JSC::Lexer<CharacterType>::parseIdentifierSlowCase):
(JSC::Lexer<T>::parseTemplateLiteral):
* Source/JavaScriptCore/parser/Lexer.h:
(JSC::Lexer::append16):
* Source/JavaScriptCore/runtime/IntlLocale.cpp:
(JSC::LocaleIDBuilder::overrideLanguageScriptRegion):
* Source/JavaScriptCore/runtime/IntlObject.cpp:
(JSC::canonicalizeUnicodeExtensionsAfterICULocaleCanonicalization):
(JSC::addScriptlessLocaleIfNeeded):
* Source/JavaScriptCore/runtime/StringPrototype.cpp:
(JSC::illFormedIndex):
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/tools/JSDollarVM.cpp:
* Source/JavaScriptCore/wasm/WasmBBQPlan.h:
* Source/JavaScriptCore/wasm/WasmEntryPlan.cpp:
(JSC::Wasm::EntryPlan::parseAndValidateModule):
* Source/JavaScriptCore/wasm/WasmEntryPlan.h:
* Source/JavaScriptCore/wasm/WasmIPIntPlan.cpp:
(JSC::Wasm::IPIntPlan::IPIntPlan):
* Source/JavaScriptCore/wasm/WasmLLIntPlan.cpp:
(JSC::Wasm::LLIntPlan::LLIntPlan):
* Source/JavaScriptCore/wasm/WasmStreamingCompiler.h:
* Source/JavaScriptCore/wasm/WasmStreamingParser.cpp:
(JSC::Wasm::StreamingParser::consume):
(JSC::Wasm::StreamingParser::consumeVarUInt32):
(JSC::Wasm::StreamingParser::addBytes):
(JSC::Wasm::StreamingParser::finalize):
* Source/JavaScriptCore/wasm/WasmStreamingParser.h:
(JSC::Wasm::StreamingParser::addBytes):
* Source/WTF/wtf/StreamBuffer.h:
(WTF::StreamBuffer::append):
* Source/WTF/wtf/TrailingArray.h:
(WTF::TrailingArray::span const):
* Source/WTF/wtf/URL.cpp:
(WTF::appendEncodedHostname):
* Source/WTF/wtf/URLParser.cpp:
(WTF::URLParser::appendToASCIIBuffer):
(WTF::URLParser::utf8PercentEncode):
(WTF::URLParser::utf8QueryEncode):
(WTF::URLParser::copyASCIIStringUntil):
(WTF::URLParser::syntaxViolation):
(WTF::URLParser::parse):
(WTF::URLParser::appendNumberToASCIIBuffer):
(WTF::URLParser::serializeIPv6):
(WTF::URLParser::domainToASCII):
(WTF::URLParser::parseHostAndPort):
* Source/WTF/wtf/URLParser.h:
(WTF::URLParser::appendToASCIIBuffer): Deleted.
* Source/WTF/wtf/Vector.h:
(WTF::Vector::subspan const):
(WTF::Vector::tryAppend):
(WTF::Vector::append):
(WTF::Vector::appendList):
(WTF::Malloc>::append):
(WTF::Malloc>::appendVector):
* Source/WTF/wtf/cf/VectorCF.h:
* Source/WTF/wtf/cocoa/FileSystemCocoa.mm:
(WTF::FileSystemImpl::openTemporaryFile):
* Source/WTF/wtf/text/AtomString.h:
* Source/WTF/wtf/text/SuperFastHash.h:
(WTF::SuperFastHash::addCharacters):
(WTF::SuperFastHash::computeHash):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::String::charactersWithoutNullTermination const):
* Source/WebCore/Modules/indexeddb/server/IDBSerialization.cpp:
(WebCore::writeLittleEndian):
(WebCore::encodeKey):
(WebCore::decodeKey):
* Source/WebCore/Modules/mediastream/SFrameUtils.cpp:
(WebCore::toRbsp):
(WebCore::computeVP8PrefixBuffer):
* Source/WebCore/Modules/mediastream/STUNMessageParsing.cpp:
(WebCore::WebRTC::getSTUNOrTURNMessageLengths):
(WebCore::WebRTC::extractSTUNOrTURNMessages):
* Source/WebCore/Modules/mediastream/STUNMessageParsing.h:
* Source/WebCore/Modules/webauthn/cbor/CBORReader.cpp:
(cbor::CBORReader::readBytes):
* Source/WebCore/Modules/webauthn/cbor/CBORWriter.cpp:
(cbor::CBORWriter::encodeCBOR):
* Source/WebCore/Modules/webauthn/fido/FidoHidPacket.cpp:
(fido::FidoHidInitPacket::getSerializedData const):
(fido::FidoHidContinuationPacket::getSerializedData const):
* Source/WebCore/Modules/webauthn/fido/FidoParsingUtils.cpp:
(fido::getInitPacketData):
(fido::getContinuationPacketData):
* Source/WebCore/Modules/webauthn/fido/U2fCommandConstructor.cpp:
(fido::WebCore::constructU2fSignCommand):
* Source/WebCore/Modules/websockets/WebSocketFrame.cpp:
(WebCore::WebSocketFrame::makeFrameData):
* Source/WebCore/Modules/websockets/WebSocketHandshake.cpp:
(WebCore::WebSocketHandshake::readHTTPHeaders):
* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(PAL::appendDecimal):
* Source/WebCore/PAL/pal/text/TextCodecICU.cpp:
(PAL::TextCodecICU::encode const):
* Source/WebCore/PAL/pal/text/TextCodecLatin1.cpp:
(PAL::encodeComplexWindowsLatin1):
* Source/WebCore/PAL/pal/text/TextCodecUserDefined.cpp:
(PAL::encodeComplexUserDefined):
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::supportsPressAction const):
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp:
(WebCore::handleResponseOnStreamingAction):
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::writeLittleEndian):
(WebCore::writeLittleEndian<uint8_t>):
(WebCore::CloneSerializer::serialize):
(WebCore::CloneSerializer::dumpImageBitmap):
(WebCore::CloneSerializer::dumpIfTerminal):
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::read):
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::SubtleCrypto::wrapKey):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmECDSAMac.cpp:
(WebCore::signECDSA):
(WebCore::verifyECDSA):
* Source/WebCore/crypto/cocoa/CryptoKeyECMac.cpp:
(WebCore::CryptoKeyEC::platformExportSpki const):
(WebCore::CryptoKeyEC::platformImportPkcs8):
(WebCore::CryptoKeyEC::platformExportPkcs8 const):
* Source/WebCore/crypto/cocoa/CryptoKeyOKPCocoa.cpp:
(WebCore::CryptoKeyOKP::exportSpki const):
(WebCore::CryptoKeyOKP::exportPkcs8 const):
* Source/WebCore/crypto/cocoa/CryptoKeyRSAMac.cpp:
(WebCore::CryptoKeyRSA::exportSpki const):
(WebCore::CryptoKeyRSA::exportPkcs8 const):
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCbcCfbParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesCtrParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmAesGcmParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmHkdfParams.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmPbkdf2Params.h:
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaKeyGenParams.h:
(WebCore::CryptoAlgorithmRsaKeyGenParams::publicExponentVector const):
* Source/WebCore/crypto/parameters/CryptoAlgorithmRsaOaepParams.h:
* Source/WebCore/dom/DatasetDOMStringMap.cpp:
(WebCore::convertPropertyNameToAttributeName):
* Source/WebCore/fileapi/BlobBuilder.cpp:
(WebCore::BlobBuilder::append):
* Source/WebCore/history/HistoryItem.cpp:
(WebCore::HistoryItem::showTreeWithIndent const):
* Source/WebCore/html/parser/HTMLToken.h:
(WebCore::HTMLToken::appendToComment):
* Source/WebCore/loader/TextResourceDecoder.cpp:
(WebCore::TextResourceDecoder::decode):
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::parseEventStreamLine):
* Source/WebCore/page/PageConsoleClient.cpp:
(WebCore::PageConsoleClient::addMessage):
(WebCore::PageConsoleClient::messageWithTypeAndLevel):
* Source/WebCore/platform/SharedBuffer.cpp:
(WebCore::combineSegmentsData):
(WebCore::FragmentedSharedBuffer::getContiguousData const):
(WebCore::FragmentedSharedBuffer::read const):
* Source/WebCore/platform/SharedStringHash.cpp:
(WebCore::findSlashDotDotSlash):
(WebCore::findSlashSlash):
(WebCore::findSlashDotSlash):
(WebCore::containsColonSlashSlash):
(WebCore::cleanSlashDotDotSlashes):
(WebCore::mergeDoubleSlashes):
(WebCore::cleanSlashDotSlashes):
(WebCore::cleanPath):
(WebCore::needsTrailingSlash):
(WebCore::computeSharedStringHashInline):
(WebCore::computeSharedStringHash):
(WebCore::computeVisitedLinkHash):
* Source/WebCore/platform/SharedStringHash.h:
* Source/WebCore/platform/audio/MultiChannelResampler.cpp:
(WebCore::MultiChannelResampler::provideInputForChannel):
* Source/WebCore/platform/audio/SincResampler.cpp:
(WebCore::SincResampler::processBuffer):
(WebCore::SincResampler::process):
* Source/WebCore/platform/graphics/cg/ImageBufferUtilitiesCG.cpp:
(WebCore::encode):
(WebCore::encodeToVector):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::WebMParser::AudioTrackData::consumeFrameData):
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.h:
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.mm:
(WebCore::cookieFromVorbisCodecPrivate):
(WebCore::createVorbisAudioInfo):
* Source/WebCore/platform/network/FormData.cpp:
(WebCore::FormData::flatten const):
* Source/WebCore/platform/network/FormDataBuilder.cpp:
(WebCore::FormDataBuilder::append):
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::parseHTTPHeader):
(WebCore::parseHTTPRequestBody):
* Source/WebCore/platform/network/HTTPParsers.h:
* Source/WebCore/platform/network/SynchronousLoaderClient.cpp:
(WebCore::SynchronousLoaderClient::didReceiveData):
* Source/WebCore/platform/network/curl/CurlMultipartHandle.cpp:
(WebCore::CurlMultipartHandle::parseHeadersIfPossible):
* Source/WebCore/platform/text/QuotedPrintable.cpp:
(WebCore::quotedPrintableEncode):
* Source/WebCore/rendering/svg/SVGTextChunk.cpp:
(WebCore::SVGTextChunk::SVGTextChunk):
* Source/WebCore/svg/SVGToOTFFontConversion.cpp:
(WebCore::SVGToOTFFontConverter::appendOS2Table):
* Source/WebCore/testing/MockContentFilter.cpp:
(WebCore::MockContentFilter::maybeDetermineStatus):
* Source/WebCore/xml/parser/XMLDocumentParser.h:
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
(WebCore::PendingCallbacks::appendCharactersCallback):
(WebCore::XMLDocumentParser::characters):
(WebCore::charactersHandler):
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::sendReplyToSynchronousRequest):
* Source/WebKit/NetworkProcess/webrtc/LibWebRTCSocketClient.cpp:
(WebKit::LibWebRTCSocketClient::sendTo):
* Source/WebKit/NetworkProcess/webrtc/LibWebRTCSocketClient.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::sendToSocket):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.mm:
(WebKit::NetworkRTCTCPSocketCocoa::createMessageBuffer):
(WebKit::NetworkRTCTCPSocketCocoa::sendTo):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
(WebKit::NetworkRTCUDPSocketCocoa::sendTo):
(WebKit::NetworkRTCUDPSocketCocoaConnections::sendTo):
* Source/WebKit/Shared/API/APIData.h:
(API::Data::data const):
(API::Data::bytes const):
(API::Data::dataReference const): Deleted.
* Source/WebKit/Shared/API/APIData.serialization.in:
* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
(-[WKRemoteObjectDecoder decodeBytesForKey:returnedLength:]):
* Source/WebKit/Shared/API/c/WKData.cpp:
(WKDataGetBytes):
* Source/WebKit/Shared/APIWebArchive.mm:
(API::WebArchive::WebArchive):
* Source/WebKit/Shared/APIWebArchiveResource.mm:
(API::WebArchiveResource::WebArchiveResource):
* Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm:
(WebKit::SandboxExtension::createHandleForTemporaryFile):
* Source/WebKit/Shared/Cocoa/WKNSData.mm:
(-[WKNSData bytes]):
* Source/WebKit/Shared/mac/AuxiliaryProcessMac.mm:
(WebKit::fileContents):
(WebKit::compileAndCacheSandboxProfile):
* Source/WebKit/UIProcess/API/C/WKNotificationManager.cpp:
(WKNotificationManagerProviderDidClickNotification_b):
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageLoadData):
(WKPageLoadDataWithUserData):
(restoreFromSessionState):
* Source/WebKit/UIProcess/API/C/WKSessionStateRef.cpp:
(WKSessionStateCreateFromData):
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.cpp:
(WebKit::WebNotificationManagerProxy::providerDidCloseNotifications):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didChooseFilesForOpenPanelWithDisplayStringAndIcon):
(WebKit::WebPageProxy::getWebCryptoMasterKey):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resumeDownload):
* Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp:
(WebKit::HistoryEntryDataDecoder::operator>>):
* Source/WebKit/UIProcess/mac/WKPrintingView.mm:
(-[WKPrintingView _preparePDFDataForPrintingOnSecondaryThread]):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundlePageEditorClient.cpp:
(WebKit::InjectedBundlePageEditorClient::getPasteboardDataForRange):
* Source/WebKit/WebProcess/InjectedBundle/mac/InjectedBundleMac.mm:
(WebKit::createUnarchiver):
* Source/WebKit/WebProcess/Plugins/PDF/PDFIncrementalLoader.mm:
(WebKit::ByteRangeRequest::addData):
(WebKit::PDFPluginStreamLoaderClient::didReceiveData):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::completeTextManipulation):
* Source/WebKitLegacy/WebCoreSupport/SocketStreamHandleClient.h:
* Source/WebKitLegacy/WebCoreSupport/SocketStreamHandleImpl.cpp:
(WebCore::removeTerminationCharacters):
(WebCore::cookieDataForHandshake):
(WebCore::SocketStreamHandleImpl::platformSendHandshake):
* Source/WebKitLegacy/WebCoreSupport/SocketStreamHandleImplCFNet.cpp:
(WebCore::SocketStreamHandleImpl::readStreamCallback):
* Source/WebKitLegacy/WebCoreSupport/WebCryptoClient.mm:
(WebCryptoClient::wrapCryptoKey const):
(WebCryptoClient::unwrapCryptoKey const):
* Source/WebKitLegacy/WebCoreSupport/WebSocketChannel.cpp:
(WebCore::WebSocketChannel::didReceiveSocketStreamData):
(WebCore::WebSocketChannel::appendToBuffer):
* Source/WebKitLegacy/WebCoreSupport/WebSocketChannel.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebVisitedLinkStore.mm:
(WebVisitedLinkStore::addVisitedLink):
* Tools/DumpRenderTree/PixelDumpSupport.cpp:
(convertChecksumToPNGComment):
* Tools/TestWebKitAPI/Tests/WebCore/ASN1Utilities.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/CtapPinTest.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/CtapRequestTest.cpp:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebCore/CtapResponseTest.cpp:
(TestWebKitAPI::getTestAttestedCredentialDataBytes):
(TestWebKitAPI::getTestAttestationObjectBytes):
(TestWebKitAPI::getTestCorruptedSignResponse):
(TestWebKitAPI::getTestU2fRegisterResponse):
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/MediaLoading.mm:
(TestWebKitAPI::testVideoBytes):
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsiteDataStoreCustomPaths.mm:
(respondToRangeRequests):
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:
(TestWebKitAPI::vectorFromData):
(TestWebKitAPI::appendUTF8ToVector):
(TestWebKitAPI::H2::Connection::receive const):
* Tools/WebKitTestRunner/PixelDumpSupport.cpp:
(convertChecksumToPNGComment):

Canonical link: https://commits.webkit.org/276319@main
